smdebug/xgboost/hook.py

```diff
@@ -119,6 +121,15 @@ def _get_num_workers(self):
     def _get_worker_name(self):
         return "worker_{}".format(xgb.rabit.get_rank())
 
+    def has_default_hook_configuration(self):
+        if not self.prepared_collections:
+            self._prepare_collections()
+
+        collections_being_saved = [x.name for x in self._collections_to_save]
+        if set(collections_being_saved) == set(DEFAULT_INCLUDE_COLLECTIONS):
+            return True
+        return False
```

Comment:
Good point. I simply copied this format from the base hook + KerasHook so I'll update the implementations there too for consistency.

Reply:
Done. Refactored this in the base hook's `has_default_hook_configuration` and updated KerasHook and XGBoost to use this instead of duplicating the implementation in a separate method.